apps/ai-content-generator/src/hooks/dialog/useAIFeature.tsx

```diff
@@ -0,0 +1,17 @@
+import featureConfig, { AIFeature, AIFeatureType } from '@configs/features/featureConfig';
+import { FeatureComponentProps } from '@configs/features/featureTypes';
+import { FC } from 'react';
+
+const useAIFeature = (aiFeature?: AIFeature): FC<FeatureComponentProps> | null => {
```

Comment:
This does not need to be a hook and it doesn't follow the hook conventions - Hooks are mainly used to abstracting out useEffects/useMemos/other native hooks. This is just a regular old switch case function that can just be part of the base dialog